compiler/ccgtypes.nim

```diff
@@ -564,10 +566,19 @@ proc genRecordFieldsAux(m: BModule; n: PNode,
           result.addf("\t$1$3 $2;\n", [getTypeDescAux(m, field.loc.t, check, skField), sname, noAlias])
   else: internalError(m.config, n.info, "genRecordFieldsAux()")
 
+proc genVirtualProcHeader(m: BModule; prc: PSym; result: var Rope; asPtr: bool = false, isFwdDecl:bool = false)
+
 proc getRecordFields(m: BModule; typ: PType, check: var IntSet): Rope =
   result = newRopeAppender()
   genRecordFieldsAux(m, typ.n, typ, check, result)
-
+  if typ.itemId in m.g.graph.virtualProcsPerType:
+    let procs = m.g.graph.virtualProcsPerType[typ.itemId]
+    for prc in procs:
+      if "#->" in prc.loc.r or "#." in prc.loc.r: continue # already generated     
```

Comment:
What?

Reply:
This is related to the question I asked you a few days ago. It reaches this point when a call to the function is made from a derived type, after it was produced. I don’t get why it needs to generate the header again but will try to look for a better check. Suggestions welcome.

Reply:
As you see in trace below it comes from `genPatternCall` to my (limited) knowledge there is no other way to prevent it. Actually there is a similar test right before it 

`pat.contains({'#', '(', '@', '\''}):`

Thoughts?

```
/Volumes/Store/Nim/compiler/nim.nim(167) nim
/Volumes/Store/Nim/compiler/nim.nim(122) handleCmdLine
/Volumes/Store/Nim/compiler/main.nim(307) mainCommand
/Volumes/Store/Nim/compiler/main.nim(277) compileToBackend
/Volumes/Store/Nim/compiler/main.nim(138) commandCompileToC
/Volumes/Store/Nim/compiler/pipelines.nim(294) compilePipelineProject
/Volumes/Store/Nim/compiler/pipelines.nim(224) compilePipelineModule
/Volumes/Store/Nim/compiler/pipelines.nim(172) processPipelineModule
/Volumes/Store/Nim/compiler/pipelines.nim(24) processPipeline
/Volumes/Store/Nim/compiler/cgen.nim(2080) genTopLevelStmt
/Volumes/Store/Nim/compiler/cgen.nim(1120) genProcBody
/Volumes/Store/Nim/compiler/ccgstmts.nim(1645) genStmts
/Volumes/Store/Nim/compiler/ccgexprs.nim(3061) expr
/Volumes/Store/Nim/compiler/ccgexprs.nim(2755) genStmtList
/Volumes/Store/Nim/compiler/ccgstmts.nim(1645) genStmts
/Volumes/Store/Nim/compiler/ccgexprs.nim(3061) expr
/Volumes/Store/Nim/compiler/ccgexprs.nim(2744) genStmtList
/Volumes/Store/Nim/compiler/ccgstmts.nim(1645) genStmts
/Volumes/Store/Nim/compiler/ccgexprs.nim(3061) expr
/Volumes/Store/Nim/compiler/ccgexprs.nim(2755) genStmtList
/Volumes/Store/Nim/compiler/ccgstmts.nim(1645) genStmts
/Volumes/Store/Nim/compiler/ccgexprs.nim(3081) expr
/Volumes/Store/Nim/compiler/ccgstmts.nim(413) genVarStmt
/Volumes/Store/Nim/compiler/ccgstmts.nim(395) genSingleVar
/Volumes/Store/Nim/compiler/ccgstmts.nim(382) genSingleVar
/Volumes/Store/Nim/compiler/ccgstmts.nim(135) loadInto
/Volumes/Store/Nim/compiler/ccgcalls.nim(827) genAsgnCall
/Volumes/Store/Nim/compiler/ccgcalls.nim(695) genInfixCall
/Volumes/Store/Nim/lib/std/assertions.nim(673) genPatternCall
/Volumes/Store/Nim/compiler/ccgtypes.nim(951) getTypeDesc
/Volumes/Store/Nim/compiler/ast.nim(2052) getTypeDescAux
/Volumes/Store/Nim/compiler/ccgtypes.nim(650) getRecordDesc
/Volumes/Store/Nim/compiler/ast.nim(585) getRecordFields
```